### PR TITLE
pin-1385: Read model

### PR DESCRIPTION
**Pull request by galales**
Note: the cqrs commons module has been updated in order to include the `MultiAction`

### CODE REVIEW

File: models/src/main/scala/it/pagopa/interop/catalogmanagement/model/persistence/JsonFormats.scala [hunk 1]
```diff
@@ -0,0 +1,71 @@
...
+  implicit val cavFormat: RootJsonFormat[CatalogAttributeValue] = jsonFormat2(CatalogAttributeValue.apply)
+  implicit val saFormat: RootJsonFormat[SingleAttribute]        = jsonFormat1(SingleAttribute.apply)
+  implicit val gaFormat: RootJsonFormat[GroupAttribute]         = jsonFormat1(GroupAttribute.apply)
+
+  implicit val caFormat: RootJsonFormat[CatalogAttribute] =
+    new RootJsonFormat[CatalogAttribute] {
+      override def read(json: JsValue): CatalogAttribute = json match {
+        case JsObject(fields) =>
+          fields
+            .get("ids")
```
Comment (galales) [inline models/src/main/scala/it/pagopa/interop/catalogmanagement/model/persistence/JsonFormats.scala]: awful.
Open to suggestions 😂

File: src/main/resources/interface-specification.yml [hunk 2]
```diff
@@ -858,6 +860,11 @@ components:
           type: string
         path:
           type: string
+        checksum:
+          type: string
+        uploadDate:
+          type: string
+          format: date-time
```
Comment (galales) [inline src/main/resources/interface-specification.yml:867]: this makes easier testing the read model, but anyway it seems awkward to have a management service that does not expose data

Comment (beetlecrunch) [reply src/main/resources/interface-specification.yml:867]: Ok sounds reasonable.

File: models/src/main/scala/it/pagopa/interop/catalogmanagement/model/persistence/JsonFormats.scala (hunk 1 above)
Comment (beetlecrunch) [reply models/src/main/scala/it/pagopa/interop/catalogmanagement/model/persistence/JsonFormats.scala]: Uhmmm no revolutionary idea, maybe just for readability I will split that case match like:
```
  implicit val caFormat: RootJsonFormat[CatalogAttribute] =
    new RootJsonFormat[CatalogAttribute] {
      override def read(json: JsValue): CatalogAttribute = json match {
        case JsObject(fields) if fields.contains("ids") => gaFormat.read(json)
        case JsObject(fields) if fields.contains("id")  => saFormat.read(json)
        case other => deserializationError(s"Unable to deserialize json as a CatalogAttribute: $other")
      }

      override def write(obj: CatalogAttribute): JsValue = obj match {
        case ca: SingleAttribute => ca.toJson
        case ca: GroupAttribute  => ca.toJson
      }
    }
```

**Review by beetlecrunch — APPROVED** [review]

File: src/main/scala/it/pagopa/interop/catalogmanagement/api/impl/EServiceApiMarshallerImpl.scala [hunk 3]
```diff
@@ -6,7 +6,7 @@ import akka.http.scaladsl.unmarshalling.FromEntityUnmarshaller
 import it.pagopa.interop.catalogmanagement.api.EServiceApiMarshaller
 import it.pagopa.interop.catalogmanagement.model._
 import spray.json._
-final object EServiceApiMarshallerImpl extends EServiceApiMarshaller with SprayJsonSupport with DefaultJsonProtocol {
+object EServiceApiMarshallerImpl extends EServiceApiMarshaller with SprayJsonSupport with DefaultJsonProtocol {
```
Comment (TonioGela) [inline src/main/scala/it/pagopa/interop/catalogmanagement/api/impl/EServiceApiMarshallerImpl.scala:9]: I'm curious why this happened

**Review by TonioGela — APPROVED** [review]
The PR is immense, at a first glance everything seems okay

File: models/src/main/scala/it/pagopa/interop/catalogmanagement/model/persistence/JsonFormats.scala (hunk 1 above)
Comment (galales) [reply models/src/main/scala/it/pagopa/interop/catalogmanagement/model/persistence/JsonFormats.scala]: I was thinking of something that does not relies only on those fields names (even if I don't think is possible 🤔 ), but this is just a great improvement, I'll add it, thanks!

**Pull request merged**